File: backends/python/fixtures/test_sample.py
"""Test Python file for LSP MCP testing."""

def greet(name: str) -> str:
    """Greet a person by name."""
    return f"Hello, {name}!"

class Calculator:
    """A simple calculator class."""

    def add(self, a: int, b: int) -> int:
        """Add two numbers."""
        return a + b

    def subtract(self, a: int, b: int) -> int:
        """Subtract b from a."""
        return a - b


if __name__ == "__main__":
    calc = Calculator()
    print(greet("World"))
    print(calc.add(1, 2))

File: backends/typescript/fixtures/test_sample.ts
/**
 * Test TypeScript file for LSP MCP testing.
 */

export function greet(name: string): string {
  return `Hello, ${name}!`;
}

export class Calculator {
  add(a: number, b: number): number {
    return a + b;
  }

  subtract(a: number, b: number): number {
    return a - b;
  }
}

const calc = new Calculator();
console.log(greet("World"));
console.log(calc.add(1, 2));
